lints/lint_sub_ca_eku_valid_fields.go

```diff
@@ -0,0 +1,41 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type subCAEKUValidFields struct {
+	// Internal data here
+}
+
+func (l *subCAEKUValidFields) Initialize() error {
+	return nil
+}
+
+func (l *subCAEKUValidFields) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubCA(c) && util.IsExtInCert(c, util.EkuSynOid)
+}
+
+func (l *subCAEKUValidFields) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, ekuValue := range c.ExtKeyUsage {
+		if ekuValue == x509.ExtKeyUsageServerAuth ||
+			ekuValue == x509.ExtKeyUsageClientAuth {
+			continue
+		} else {
+			return ResultStruct{Result: Notice}, nil
```

Comment:
Blagh. 

The way this is currently written, we return a pass if either of the fields are there for being technically constrained, and a notice otherwise. 

Since notices aren't really errors, they're just interesting things to watch for, I'm okay with the lint as is. I'm also okay not having it at all, considering it doesn't tell us anything AND that a previous run of ZLint with this lint told me this never happens.